### PR TITLE
Add cookbook for accessing additional_properties

### PR DESCRIPTION
**Pull request by chutten**

### CODE REVIEW

File: src/cookbooks/additional_props.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+Until the problem is fixed, any ingested pings with these
+"unknown" data in them will place that data in the `additional_properties` column.
+And unless you convince a Data Engineer to backfill,
+these rows of the dataset will always have these data in the
+`additional_properties` column,
+even after the dataset has been updated to add the column.
+
+To access data that's been put into the `additional_properties` column,
+you'll have to parse the JSON.
```
Comment (jklukas) [inline src/cookbooks/additional_props.md:23]:
```suggestion
you'll have to parse the JSON.
Be aware that when you access the `additional_properties` field, BigQuery has
to read the entire contents, even if you're extracting just a single field.
In the case of `main` pings, `additional_properties` can be quite large, leading
to expensive queries.
```

File: src/SUMMARY.md [hunk 2]
```diff
@@ -37,6 +37,7 @@
       - [Daily Active Users (DAU and MAU)](cookbooks/dau.md)
       - [Active DAU (aDAU)](cookbooks/active_dau.md)
       - [Retention](cookbooks/retention.md)
+    - [Additional Properties](cookbooks/additional_props.md)
```
Comment (wlach) [inline src/SUMMARY.md:40]: Could you put this under "Accessing and working with BigQuery"? (last item)

```suggestion
    - [Accessing Additional Properties](cookbooks/additional_props.md)
```

(also, it looks capitalization here is a mess, but that's not your fault and we can leave it for later) 😄

File: src/cookbooks/additional_props.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+If some field is present in a valid received [ping](../concepts/terminology.md#ping),
+but is not present in the ping's [schema](../concepts/terminology.md#schema),
+it doesn't have its own column to be placed into during
+[ingestion](../concepts/terminology.md#ingestion).
+Instead, those fields remain as raw JSON and are placed in the
+`additional_properties` column of the ping's table or view.
+
+This can happen for a variety of usually-temporary problems like:
+
+- The latest schema hasn't yet been deployed
```
Comment (wlach) [inline src/cookbooks/additional_props.md:12]:
```suggestion
- The latest schema hasn't yet been deployed (see ["What does it mean when a schema deploy is blocked?"](../concepts/pipeline/schemas.html#what-does-it-mean-when-a-schema-deploy-is-blocked))
```

File: src/cookbooks/additional_props.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+- There is a bug in sending or receiving data
```
Comment (wlach) [inline src/cookbooks/additional_props.md:13]:
```suggestion
- [mozilla-pipeline-schemas](https://github.com/mozilla-services/mozilla-pipeline-schemas/commit/0a24af0b56b6645906858ed02b20f977a4fb3842) was not updated after a change in the data which was submitted (this shouldn't happen with [Glean](../concepts/glean.md), but can happen with some legacy data, for example the legacy desktop telemetry envionment).
```

**Review by wlach — CHANGES_REQUESTED** [review]
Have some minor requested changes, but in general this looks great! Thanks for taking the time to do it.

File: src/cookbooks/additional_props.md (hunk 4 above)
Comment (chutten) [reply src/cookbooks/additional_props.md:13]: Why this commit in particular? (Also, I'd specify "legacy Firefox Desktop telemetry environment")

Comment (wlach) [reply src/cookbooks/additional_props.md:13]: Eugh, I did not mean to link a specific commit there. Sorry, long day :)

File: src/cookbooks/additional_props.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+If some field is present in a valid received [ping](../concepts/terminology.md#ping),
+but is not present in the ping's [schema](../concepts/terminology.md#schema),
+it doesn't have its own column to be placed into during
+[ingestion](../concepts/terminology.md#ingestion).
+Instead, those fields remain as raw JSON and are placed in the
+`additional_properties` column of the ping's table or view.
+
+This can happen for a variety of usually-temporary problems like:
+
+- The latest schema hasn't yet been deployed (see ["What does it mean when a schema deploy is blocked?"](../concepts/pipeline/schemas.html#what-does-it-mean-when-a-schema-deploy-is-blocked))
```
Comment (fbertsch) [inline src/cookbooks/additional_props.md:12]: It is important to note that we do not backfill schema changes. So when the new column gets added, new data will flow there, and the existing query will be wrong.

File: src/cookbooks/additional_props.md (hunk 4 above)
Comment (wlach) [reply src/cookbooks/additional_props.md:13]: Adjusted the suggestion

File: src/cookbooks/additional_props.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+- [mozilla-pipeline-schemas](https://github.com/mozilla-services/mozilla-pipeline-schemas) was not updated after a change in the data which was submitted (this shouldn't happen with [Glean](../concepts/glean.md), but can happen with some legacy data, for example the legacy desktop telemetry envionment).
```
Comment (chutten) [inline src/cookbooks/additional_props.md:13]:
```suggestion
- [mozilla-pipeline-schemas](https://github.com/mozilla-services/mozilla-pipeline-schemas) was not updated after a change in the data which was submitted (this shouldn't happen with [Glean](../concepts/glean.md), but can happen with some legacy data, for example the legacy Firefox Desktop telemetry environment).
```

Comment (wlach) [inline src/cookbooks/additional_props.md:13]: Thanks for addressing this! It also occurs to me that we could link to a reference on what the telemetry environment is.

```suggestion
- [mozilla-pipeline-schemas](https://github.com/mozilla-services/mozilla-pipeline-schemas) was not updated after a change in the data which was submitted (this shouldn't happen with [Glean](../concepts/glean.md), but can happen with some legacy data, for example the [legacy desktop telemetry environment](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/data/environment.html)).
```

**Review by wlach — APPROVED** [review]

**Comment by chutten** [comment]
Oh bother, we seem to have mid-aired.

File: src/cookbooks/additional_props.md (hunk 6 above)
Comment (chutten) [reply src/cookbooks/additional_props.md]: Can't apply outdated suggestions : |

File: src/cookbooks/additional_props.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+but is not present in the ping's [schema](../concepts/terminology.md#schema),
+it doesn't have its own column to be placed into during
+[ingestion](../concepts/terminology.md#ingestion).
+Instead, those fields remain as raw JSON and are placed in the
+`additional_properties` column of the ping's table or view.
+
+This can happen for a variety of usually-temporary problems like:
+
+- The latest schema hasn't yet been deployed (see ["What does it mean when a schema deploy is blocked?"](../concepts/pipeline/schemas.html#what-does-it-mean-when-a-schema-deploy-is-blocked))
+- [mozilla-pipeline-schemas](https://github.com/mozilla-services/mozilla-pipeline-schemas) was not updated after a change in the data which was submitted (this shouldn't happen with [Glean](../concepts/glean.md), but can happen with some legacy data, for example the legacy Firefox Desktop telemetry environment).
```
Comment (chutten) [inline src/cookbooks/additional_props.md:13]:
```suggestion
- [`mozilla-pipeline-schemas`](https://github.com/mozilla-services/mozilla-pipeline-schemas) was not updated after a change in the data which was submitted (this shouldn't happen with [Glean](../concepts/glean.md), but can happen with some legacy data, for example the [legacy Firefox Desktop telemetry environment](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/data/environment.html)).
```

File: src/cookbooks/additional_props.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+This can happen for a variety of usually-temporary problems like:
+
+- The latest schema hasn't yet been deployed (see ["What does it mean when a schema deploy is blocked?"](../concepts/pipeline/schemas.html#what-does-it-mean-when-a-schema-deploy-is-blocked))
+- [`mozilla-pipeline-schemas`](https://github.com/mozilla-services/mozilla-pipeline-schemas) was not updated after a change in the data which was submitted (this shouldn't happen with [Glean](../concepts/glean.md), but can happen with some legacy data, for example the [legacy Firefox Desktop telemetry environment](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/data/environment.html)).
+
+Until the problem is fixed, any ingested pings with these
+"unknown" data in them will place that data in the `additional_properties` column.
+And unless you convince a Data Engineer to backfill,
+these rows of the dataset will always have these data in the
```
Comment (fbertsch) [inline src/cookbooks/additional_props.md:18]:
```suggestion
The rows of this dataset will always have these data in the
```

I'd rather not have people try and convince us to backfill data for them :)

File: src/cookbooks/additional_props.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+but is not present in the ping's [schema](../concepts/terminology.md#schema),
+it doesn't have its own column to be placed into during
+[ingestion](../concepts/terminology.md#ingestion).
+Instead, those fields remain as raw JSON and are placed in the
+`additional_properties` column of the ping's table or view.
+
+This can happen for a variety of usually-temporary problems like:
+
+- The latest schema hasn't yet been deployed (see ["What does it mean when a schema deploy is blocked?"](../concepts/pipeline/schemas.html#what-does-it-mean-when-a-schema-deploy-is-blocked))
+- [`mozilla-pipeline-schemas`](https://github.com/mozilla-services/mozilla-pipeline-schemas) was not updated after a change in the data which was submitted (this shouldn't happen with [Glean](../concepts/glean.md), but can happen with some legacy data, for example the [legacy Firefox Desktop telemetry environment](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/data/environment.html)).
```
Comment (chutten) [inline src/cookbooks/additional_props.md:13]:
```suggestion
- [`mozilla-pipeline-schemas`](https://github.com/mozilla-services/mozilla-pipeline-schemas) was not updated after a change in the data which was submitted (this shouldn't happen with [Glean](../concepts/terminology.md#glean), but can happen with some legacy data, for example the [legacy Firefox Desktop telemetry environment](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/data/environment.html)).
```

File: src/cookbooks/additional_props.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+If some field is present in a valid received [ping](../concepts/terminology.md#ping),
+but is not present in the ping's [schema](../concepts/terminology.md#schema),
+it doesn't have its own column to be placed into during
+[ingestion](../concepts/terminology.md#ingestion).
+Instead, those fields remain as raw JSON and are placed in the
+`additional_properties` column of the ping's table or view.
+
+This can happen for a variety of usually-temporary problems like:
+
+- The latest schema hasn't yet been deployed (see ["What does it mean when a schema deploy is blocked?"](../concepts/pipeline/schemas.html#what-does-it-mean-when-a-schema-deploy-is-blocked))
```
Comment (chutten) [inline src/cookbooks/additional_props.md:12]:
```suggestion
- The latest schema hasn't yet been deployed (see ["What does it mean when a schema deploy is blocked?"](../concepts/pipeline/schemas.md#what-does-it-mean-when-a-schema-deploy-is-blocked))
```

**Comment by chutten** [comment]
Linkcheck's failures are in other files and appear benign. Going ahead with merge.

**Pull request merged**